### PR TITLE
A few cleanups on tile hashing

### PR DESCRIPTION
**Pull request by nyurik**
* Rename `global_hash` to `agg_tiles_hash`

This is still a big sticking point: what should be the name for the metadata key for this value? The value represents the hash of all `z,x,y,tile` over all rows of the `tiles` table (or view).  Should it include `md5` in its name, or should the hash be auto-detected by its length? (details in #856)

* Generate it based on `tiles` table/view
* validate or generate, but not both (it will always fail otherwise)
* break up logic for per-tile, total, and integrity checks
* delete unused sqlx prep file

### CODE REVIEW

**Comment by sharkAndshark** [comment]
If it's true each hash digest has a unique byte length, I prefer the `auto-guessing`.  
Should we allow user to choose algorithms(default `MD5`) when generating hash?

||feature|drawback|
|----|----|----|
|algorithm in name|  explain its algorithm `explicitly` | different name across mbtile files like `tiles_md5_hash` in file1, `tiles_sha1_hash` in file2  |
| guessing |  explain its algorithm `implicitly`; consistent metadata name across mbtile files||

**Comment by nyurik** [comment]
eventually we could let users pick algorithms - not very difficult, but not worth our time at the moment i think - md5 has worked pretty well for this usecase, despite being ... cryptographically poor.  I agree about autoguessing.  I think `agg_tiles_hash` sounds best, as it is both short, and implies aggregation

**Comment by sharkAndshark** [comment]
||feature|drawback|
|----|----|----|
|algorithm in name|  explain its algorithm `explicitly` | different name across mbtile files like `tiles_md5_hash` in file1, `tiles_sha1_hash` in file2  |
| guessing |  explain its algorithm `implicitly`; consistent metadata name across mbtile files||

File: martin/src/pg/pool.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ impl PgPool {
             .build()
             .map_err(|e| PostgresPoolBuildError(e, id.clone()))?;
 
-        let version = get_conn(&pool, id.as_str())
+        let version: String = get_conn(&pool, id.as_str())
```
Comment (sharkAndshark) [inline martin/src/pg/pool.rs]: Seems there's few connection between cleaning up on `pg/pool.rs` and the target of this PR. Not sure as I didn't do much on mbtiles part of martin.

Comment (nyurik) [reply martin/src/pg/pool.rs]: sorry, that was a minor cleanup. Fixed

**Comment by GeoKarim** [comment]
Did a quick overview, all looks good to me.

**Comment by nyurik** [comment]
I walked @GeoKarim (my coworker) through the code to make sure there are no obvious oopsies

**Pull request merged**